pre-commit.yml

```
@@ -0,0 +1,18 @@
+repos:
+- repo: https://github.com/astral-sh/ruff-pre-commit
+  rev: v0.1.14
+  hooks:
+  - id: ruff
+  - id: ruff-format
+    args: [--check]
```

```suggestion
    args: [--diff]
```

Could you also replace `format --check` with `format --diff` that in `lint.yml`?